subsys/bluetooth/host/l2cap.c

```diff
@@ -1226,7 +1222,7 @@ static struct net_buf *l2cap_chan_create_seg(struct bt_l2cap_le_chan *ch,
 
 static void l2cap_chan_tx_resume(struct bt_l2cap_le_chan *ch)
 {
-	if (!k_sem_count_get(&ch->tx.credits) ||
+	if (!atomic_get(&ch->tx.credits) ||
 	    (k_fifo_is_empty(&ch->tx_queue) && !ch->tx_buf)) {
 		return;
 	}
```

Comment:
one question: any complications here if credits gets set to 0 by another CPU in between the atomic_get() and the k_work_submit() call?

Reply:
There shouldn't be any problem since the code checks again if there any credits on the work itself, that is actually where test_and_dec is used so it would basically just defered itself again until there are credits available.

Reply:
sounds good